src/video/SDL_blit_A.c

```diff
@@ -1288,31 +1134,69 @@ static void BlitNtoNPixelAlpha(SDL_BlitInfo *info)
     SDL_PixelFormat *dstfmt = info->dst_fmt;
     int srcbpp;
     int dstbpp;
+    SDL_bool freeFormat;
     Uint32 Pixel;
     unsigned sR, sG, sB, sA;
     unsigned dR, dG, dB, dA;
 
     /* Set up some basic variables */
     srcbpp = srcfmt->BytesPerPixel;
     dstbpp = dstfmt->BytesPerPixel;
+    freeFormat = SDL_FALSE;
+
+#ifdef SDL_AVX2_INTRINSICS
+    if (srcbpp == 4 && dstbpp == 4 && width >= 4 && SDL_HasAVX2()) {
+        BlitNtoNPixelAlpha_AVX2(info);
+        return;
+    }
+#endif
+
+#ifdef SDL_SSE4_1_INTRINSICS
+	if (srcbpp == 4 && dstbpp == 4 && width >= 2 && SDL_HasSSE41()) {
+        BlitNtoNPixelAlpha_SSE4_1(info);
+	    return;
+    }
+#endif
+    /* Handle case where bad input sent */
```

Comment:
This isn't really "bad input" - it would be considered valid for pixel formats like SDL_PIXELFORMAT_XRGB8888 where the top 8 bits are simply padding instead of an alpha channel.

Reply:
I called this bad input because it's reproducible using sdl12-compat -> sdl2-compat -> SDL3 with the application Cogmind, which uses a non-standard ARBG source format and a standard ARGB format for the destination. Perhaps the correct approach is to fix what is potentially a bug in sdl2-compat instead.

I'm not sure we need to consider the correct application of blending for transport to/from XRGB8888 within this function. I believe that is handled within SDL_blit_auto's `SDL_Blit_XRGB8888_XRGB8888_Blend`, `SDL_Blit_XRGB8888_ARGB8888_Blend`, and other similar functions. I may be mistaken in my assessment however, I will debug an example blend with these formats to make sure.